token/js/src/extensions/tokenMetadata/state.ts

```diff
@@ -0,0 +1,105 @@
+import type { Commitment, Connection } from '@solana/web3.js';
+import { PublicKey } from '@solana/web3.js';
+import type { TokenMetadata } from '@solana/spl-token-metadata';
+import { Field, unpack } from '@solana/spl-token-metadata';
+
+import { TOKEN_2022_PROGRAM_ID } from '../../constants.js';
+import { ExtensionType, getExtensionData } from '../extensionType.js';
+import { getMint } from '../../state/mint.js';
+
+export const getNormalizedTokenMetadataField = (field: Field | string): string => {
+    if (field === Field.Name || field === 'Name' || field === 'name') {
+        return 'name';
+    }
+
+    if (field === Field.Symbol || field === 'Symbol' || field === 'symbol') {
+        return 'symbol';
+    }
+
+    if (field === Field.Uri || field === 'Uri' || field === 'uri') {
+        return 'uri';
+    }
+
+    return field;
+};
+
+export const updateTokenMetadata = (
```

Comment:
I don't see any ways to simply this function but i'm open to ideas - it feels a bit overkill and might be missing some edge cases. 

We only really need the change in length,  concerned if we have this we the behaviour in sync with the rust implementation. Is this an issue?. 

Alternatively we could change this to just calculate the change in data len - I didn't do it because i thought this better maps to the domain logic.